src/tools/dashboard/src/reference.rs

```diff
@@ -0,0 +1,287 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+//! Utilities to extract examples from The Rust Reference, run them through RMC,
+//! and display their results.
+
+use pulldown_cmark::{Parser, Tag};
+use std::io::{BufRead, BufReader};
+use std::process::Stdio;
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+    process::Command,
+};
+use std::{env, fs};
+
+use crate::dashboard;
+
+/// Parses the chapter/section hierarchy in the markdown file specified by
+/// `summary_path` and returns a mapping from markdown files containing rust
+/// code to corresponding directories where the extracted rust code should
+/// reside.
+fn parse_hierarchy(summary_path: &Path) -> HashMap<PathBuf, PathBuf> {
+    let summary = fs::read_to_string(summary_path).unwrap();
+    // Skip the title and introduction.
+    let parser = Parser::new(&summary).skip(8);
```

Comment:
Can we make give a name to `8` and add a check here? I'm thinking of a scenario where they update the reference and this hard-coded number consumes too little/much.

Reply:
I also like to write the actual expected string and use `.len()` to get the value rather than writing the number itself.

Reply:
@vecchiot-aws great idea!